src/screens/Recon/ReconModule.res

```diff
@@ -0,0 +1,95 @@
+@react.component
+let make = (~urlList) => {
+  open APIUtils
+  open LogicUtils
+
+  let getURL = useGetURL()
+  let fetchDetails = useGetMethod()
+  let (redirectToken, setRedirecToken) = React.useState(_ => "")
+  let (screenState, setScreenState) = React.useState(_ => PageLoaderWrapper.Loading)
+  let (iframeLoaded, setIframeLoaded) = React.useState(_ => false)
+  let iframeRef = React.useRef(Js.Nullable.null)
+
+  let getReconToken = async () => {
+    try {
+      let url = getURL(~entityName=RECON, ~reconType=#TOKEN, ~methodType=Get, ())
+      let res = await fetchDetails(url)
+      let token = res->LogicUtils.getDictFromJsonObject->LogicUtils.getString("token", "")
+      setRedirecToken(_ => token)
+      setScreenState(_ => PageLoaderWrapper.Success)
+    } catch {
+    | _ => setScreenState(_ => PageLoaderWrapper.Error("Something went wrong!"))
+    }
+  }
+
+  let redirectUrl = switch urlList {
+  | list{"upload-files"}
+  | list{"run-recon"}
+  | list{"reports"}
+  | list{"config-settings"}
+  | list{"file-processor"} =>
+    urlList->List.toArray->Array.joinWithUnsafe("/")
+  | list{"recon-analytics"} => "analytics"
+  | _ => ""
+  }
+
+  React.useEffect2(() => {
+    getReconToken()->ignore
+    None
+  }, (iframeLoaded, redirectUrl))
+
+  <>
+    {
+      switch iframeRef.current->Js.Nullable.toOption {
+      | Some(iframeEl) => {
+          let tokenDict = [("token", redirectToken->JSON.Encode.string)]->Dict.fromArray
+          let dict =
+            [
+              ("eventType", "AuthenticationDetails"->JSON.Encode.string),
+              ("payload", tokenDict->JSON.Encode.object),
+            ]->Dict.fromArray
+          iframeEl->IframeUtils.iframePostMessage(dict)
+        }
+
+      | None => ()
+      }
```

Comment:
This block doesn't need to be inside the fragment right. Can be before it

Reply:
We need to reverify because placing the iframe logic prevents the post message from being sent to the child iframe.

Reply:
As we need to deploy this in sbx , will take up the refactoring of the module in a following PR